### ISSUE TITLE
stage1.sh

### ISSUE THREAD

**Issue by CatherineYun** [SY-Xuan/Pink#6]
```
export PYTHONPATH=$PYTHONPATH:./

output_dir=./dir_satge1
if [ -d ${output_dir} ];then
    echo "dir already exists"
else
    mkdir ${output_dir}
fi
export CUDA_LAUNCH_BLOCKING=1



llama_path=./Llama-2-7b-chat-hf
llava_cc3m_pretrain_data_path=./LLaVA-CC3M-Pretrain-595K/chat.json
llava_cc3m_pretrain_base_path=./LLaVA-CC3M-Pretrain-595K/images
llava_cc3m_pretrain_image_folder=./LLaVA-CC3M-Pretrain-595K/images

OMP_NUM_THREADS=1 torchrun --nnodes=1 --nproc_per_node=4 --master_port=25002 \
    pink/train/train.py \
    --model_name_or_path ${llama_path} \
    --llama_path ${llama_path} \
    --dataset_name LLaVACaptionDataset \
    --data_path ${llava_cc3m_pretrain_data_path} \
    --image_folder ${llava_cc3m_pretrain_image_folder} \
    --base_path ${llava_cc3m_pretrain_base_path} \
    --vision_tower ./clip-vit-large-patch14 \
    --tune_mm_mlp_adapter True \
    --mm_vision_select_layer -2 \
    --conversation_template llamav2 \
    --freeze_llm True \
    --llm_adapter_enable False \
    --visual_adapter_enable False \
    --freeze_vit True \
    --bf16 True \
    --output_dir ${output_dir} \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 16 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 2400000 \
    --save_total_limit 1 \
    --learning_rate 2e-3 \
    --dataloader_num_workers 4 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing False \
    --report_to tensorboard
```
这是我的stage1.sh的内容但是会有报错
```
WARNING:accelerate.utils.other:Detected kernel version 3.10.0, which is below the recommended minimum of 5.5.0; this can cause the process to hang. It is recommended to upgrade the kernel to the minimum version or higher.
  0%|                                                                            | 0/9302 [00:00<?, ?it/s][rank3]:[W reducer.cpp:1360] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank0]:[W reducer.cpp:1360] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank2]:[W reducer.cpp:1360] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[rank1]:[W reducer.cpp:1360] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 7.142857142857143e-06, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 1.4285714285714285e-05, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 2.1428571428571428e-05, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 2.857142857142857e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 3.571428571428571e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 4.2857142857142856e-05, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 5e-05, 'epoch': 0.0}                                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 5.714285714285714e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 6.428571428571427e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 7.142857142857142e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 7.857142857142857e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 8.571428571428571e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 9.285714285714286e-05, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0001, 'epoch': 0.0}                                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00010714285714285714, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00011428571428571428, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00012142857142857143, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00012857142857142855, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0001357142857142857, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00014285714285714284, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00015, 'epoch': 0.0}                                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00015714285714285713, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00016428571428571428, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00017142857142857143, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00017857142857142857, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00018571428571428572, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00019285714285714286, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002, 'epoch': 0.0}                                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00020714285714285716, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00021428571428571427, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00022142857142857142, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00022857142857142857, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002357142857142857, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00024285714285714286, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00025, 'epoch': 0.0}                                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002571428571428571, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002642857142857143, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002714285714285714, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002785714285714286, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002857142857142857, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0002928571428571429, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0003, 'epoch': 0.0}                                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0003071428571428572, 'epoch': 0.0}                     
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00031428571428571427, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00032142857142857147, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00032857142857142856, 'epoch': 0.0}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0003357142857142857, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00034285714285714285, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00035, 'epoch': 0.01}                                  
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00035714285714285714, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0003642857142857143, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00037142857142857143, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0003785714285714286, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0003857142857142857, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0003928571428571429, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0004, 'epoch': 0.01}                                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00040714285714285717, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0004142857142857143, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00042142857142857146, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00042857142857142855, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00043571428571428575, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00044285714285714284, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00045000000000000004, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00045714285714285713, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00046428571428571433, 'epoch': 0.01}                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0004714285714285714, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0004785714285714286, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0004857142857142857, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0004928571428571429, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005, 'epoch': 0.01}                                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005071428571428572, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005142857142857142, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005214285714285714, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005285714285714286, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005357142857142857, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005428571428571428, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.00055, 'epoch': 0.01}                                  
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005571428571428572, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005642857142857143, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005714285714285714, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005785714285714286, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005857142857142858, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0005928571428571429, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006, 'epoch': 0.01}                                   
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006071428571428571, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006142857142857143, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006214285714285715, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006285714285714285, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006357142857142857, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006428571428571429, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006500000000000001, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006571428571428571, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006642857142857143, 'epoch': 0.01}                    
{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 0.0006714285714285714, 'epoch': 0.01} 
```

**Comment by SY-Xuan** [comment]
What is the version of the Transformers? It looks like your training is not correct. Maybe you can check your image loading, and weight loading? In addition, you can try to decrease the learning rate and increase the training batch size.

**Comment by CatherineYun** [comment]
transformers  版本是4.38.2 检查了一下，换了模型之后就正常了 麻烦您啦
